openedx_learning/core/contents/api.py

```diff
@@ -80,69 +40,122 @@ def get_or_create_media_type_id(mime_type: str) -> int:
+    component XBlock like a ProblemBlock.
+
+    This function will *always* create a text entry in the database. In addition
+    to this, if you specify ``create_file=True``, it will also save a copy of
+    that text data to the file storage backend. This is useful if we want to let
+    that file be downloadable by browsers in the LMS at some point.
+
+    If you want to create a large text file, or want to create a text file that
+    doesn't need to be stored in the database, call ``create_file_content``
+    instead of this function.
+
+    The `hash_digest` is included as an optional argument because a very common
+    pattern is going to be "I have this data, let's see if a Content already
+    exists for it."
     """
-    hash_digest = hash_digest or create_hash_digest(data_bytes)
-    try:
-        raw_content = RawContent.objects.get(
-            learning_package_id=learning_package_id, hash_digest=hash_digest
-        )
-        was_created = False
-    except RawContent.DoesNotExist:
-        raw_content = create_raw_content(
-            learning_package_id, data_bytes, mime_type, created, hash_digest
-        )
-        was_created = True
-
-    return raw_content, was_created
-
-
-def get_or_create_text_content_from_bytes(
+    text_as_bytes = text.encode('utf-8')
+    hash_digest = create_hash_digest(text_as_bytes)
+
+    with atomic():
+        try:
+            content = Content.objects.get(
+                learning_package_id=learning_package_id,
+                media_type_id=media_type_id,
+                hash_digest=hash_digest,
+            )
+        except Content.DoesNotExist:
+            content = Content(
+                learning_package_id=learning_package_id,
+                media_type_id=media_type_id,
+                hash_digest=hash_digest,
+                created=created,
+                size=len(text_as_bytes),
+                text=text,
+                has_file=create_file,
+            )
+            content.full_clean()
+            content.save()
+
+            if create_file:
+                content.write_file(ContentFile(text_as_bytes))
+
+        return content
+
+
+def get_or_create_file_content(
     learning_package_id: int,
     /,
-    data_bytes: bytes,
-    mime_type: str,
+    media_type_id: int,
+    data: bytes,
     created: datetime,
-    hash_digest: str | None = None,
-    encoding: str = "utf-8-sig",
-):
+) -> Content:
     """
-    Get the TextContent in the given learning package with the specified data,
-    or create it if it doesn't exist.
+    Get or create a Content with data stored in a file storage backend.
+
+    Use this function to store non-text data, large data, or data where low
+    latency access is not necessary. Also use this function to store any Content
+    that you want to be downloadable by browsers in the LMS since the static
+    asset serving system will only work with file-backed Content.
```

Comment:
suggested clarification, if you agree that this is the case:
```suggestion
    latency access is not necessary. Also use this function (or get_or_create_text_content
    with create_file=True) to store any Content that you want to be downloadable by
    browsers in the LMS since the static asset serving system will only work with file-backed Content.
```